cleanupsdemo/cleanups_demo.go

```diff
@@ -22,14 +24,24 @@ func main() {
 	// of the thunks that are registered with `runtime.SetFinalizer()` in
 	// group.go and collect.go, which close channels and call cancel functions.
 
-	println("leaking ~", cycles*(batchSize+2), "goroutines from abandoned group executors...")
+	println("leaking goroutines from group executors...")
+
+	// Dependent contexts are always canceled, too
+	leakDependent := func(ctx context.Context) {
+		ctx, cancel := context.WithCancel(ctx)
+		_ = cancel
+		go func() {
+			<-ctx.Done()
+		}()
+	}
 
 	// Leak just a crazy number of goroutines
 	for i := 0; i < cycles; i++ {
 		func() {
-			g := parallel.Collect[int](parallel.Limited(context.Background(), batchSize))
+			g := parallel.Collect[int](parallel.Limited(ctx, batchSize))
```

Comment:
The original version of this was a case where the previous version of `parallel` already did the correct thing and prevented a leak, right? I don't understand what the leak was before adding `leakDependent()`

Reply:
the previous version of the library prevents leaking forever-stalled goroutines, but does not prevent leaking uncanceled contexts. now it does both.

the difference only really manifests when an executor is awaited and completes normally and without error, a chain of events that previously would see the context in a non-errored state (so, not canceled). we've been using `context.Cause(ctx)` as the canonical store of single error values for things like `ErrGroup`, which meant that naively if we would always cancel the context then it would always look like there was an error. To close this gap we had to introduce this sentinel error value, `errGroupDone`, to represent the "context is canceled but 
everything is fine" state so we can recognize that no error occurred.